kore/src/Kore/Logger/ErrorBracket.hs

```diff
@@ -0,0 +1,100 @@
+{- |
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+-}
+module Kore.Logger.ErrorBracket
+    ( ErrorBracket (..)
+    ) where
+
+import Control.Exception
+    ( onException
+    )
+import qualified Control.Monad.State.Strict as Strict
+    ( StateT (..)
+    )
+import Control.Monad.Trans.Accum
+    ( AccumT (..)
+    , runAccumT
+    )
+import Control.Monad.Trans.Except
+    ( ExceptT (..)
+    , runExceptT
+    )
+import Control.Monad.Trans.Identity
+    ( IdentityT (..)
+    )
+import Control.Monad.Trans.Maybe
+    ( MaybeT (..)
+    )
+import Control.Monad.Trans.Reader
+    ( ReaderT (..)
+    )
+import Data.Text
+    ( Text
+    )
+import qualified Data.Text.IO as Text
+    ( putStrLn
+    )
+import GHC.Stack
+    ( HasCallStack
+    , callStack
+    , prettyCallStack
+    )
+
+import ListT
+    ( ListT
+    , mapListT
+    )
+
+class Monad m => ErrorBracket m where
+    withErrorMessage :: HasCallStack => Text -> m a -> m a
+
+instance ErrorBracket m => ErrorBracket (Strict.StateT s m) where
+    withErrorMessage message action =
+        Strict.StateT
```

Comment:
This is `Control.Monad.Morph.hoist (withErrorMessage message)`, and so are most of the instances below, but see my comment about `MonadCatch`.

Reply:
Switched to MonadCatch